### PR TITLE
Multiple fixes for PaymentIntent

### PR DESCRIPTION
**Pull request by remi-stripe**
Fixes https://github.com/stripe/stripe-dotnet/issues/1532 and other issues

r? @ob-stripe 
cc @stripe/api-libraries

### CODE REVIEW

File: src/Stripe.net/Entities/PaymentIntents/PaymentIntentTransferData.cs [hunk 1]
```diff
@@ -6,7 +6,7 @@ namespace Stripe
     public class PaymentIntentTransferData : StripeEntity
     {
         [JsonProperty("amount")]
-        public long Amount { get; set; }
+        public long? Amount { get; set; }
```
Comment (remi-stripe) [inline src/Stripe.net/Entities/PaymentIntents/PaymentIntentTransferData.cs]: Is this a breaking change?

Comment (ob-stripe) [reply src/Stripe.net/Entities/PaymentIntents/PaymentIntentTransferData.cs]: In the strictest sense, yes: code that uses this property will probably need to be updated to handle the nullability. That said, given that very few people are using this, and anyone who does is probably seeing deserialization errors, I think it's fine to consider this a bug fix and release as a patch version (though the rest of the changes in this PR qualify this as a minor, not a patch).

**Comment by remi-stripe** [comment]
@ob-stripe I made some extra changes, can you re-review and if it's all good release a version?

File: src/Stripe.net/Services/PaymentIntents/PaymentIntentSharedOptions.cs [hunk 2]
```diff
@@ -27,15 +27,18 @@ public class PaymentIntentSharedOptions : BaseOptions
...
         public string OnBehalfOf { get; set; }
 
-        [JsonProperty("payment_method_types")]
-        public List<string> PaymentMethodTypes { get; set; }
-
         [JsonProperty("receipt_email")]
         public string ReceiptEmail { get; set; }
 
+        [JsonProperty("payment_method_types")]
+        public List<string> PaymentMethodTypes { get; set; }
```
Comment (ob-stripe) [inline src/Stripe.net/Services/PaymentIntents/PaymentIntentSharedOptions.cs]: Looks like this one was already in the correct alphabetical order, can you move it back?

File: src/Stripe.net/Entities/PaymentIntents/PaymentIntentTransferData.cs [hunk 3]
```diff
@@ -29,5 +27,9 @@ internal object InternalDestination
             }
         }
         #endregion
+
+        [Obsolete("Use ApplicationFeeAmount instead")]
```
Comment (ob-stripe) [inline src/Stripe.net/Entities/PaymentIntents/PaymentIntentTransferData.cs]: The class doesn't have an `ApplicationFeeAmount` property. Can you change the message to `"Use PaymentIntent.ApplicationFeeAmount"`?

File: src/Stripe.net/Services/PaymentIntents/PaymentIntentTransferDataOptions.cs [hunk 4]
```diff
@@ -1,14 +1,16 @@
...
     public class PaymentIntentTransferDataOptions : INestedOptions
     {
-        [JsonProperty("amount")]
-        public long? Amount { get; set; }
-
         // This is only available on creation and not update
         [JsonProperty("destination")]
         public string Destination { get; set; }
+
+        [Obsolete("Use ApplicationFeeAmount instead")]
```
Comment (ob-stripe) [inline src/Stripe.net/Services/PaymentIntents/PaymentIntentTransferDataOptions.cs]: Same here, can you change the message to `"Use PaymentIntent*Options.ApplicationFeeAmount"`?

**Comment by remi-stripe** [comment]
@ob-stripe thanks for catching all the issues. PTAL.

File: src/Stripe.net/Services/PaymentIntents/PaymentIntentTransferDataOptions.cs [hunk 5]
```diff
@@ -1,14 +1,16 @@
...
     public class PaymentIntentTransferDataOptions : INestedOptions
     {
-        [JsonProperty("amount")]
-        public long? Amount { get; set; }
-
         // This is only available on creation and not update
         [JsonProperty("destination")]
         public string Destination { get; set; }
+
+        [Obsolete("Use PaymentIntent.ApplicationFeeAmount instead")]
```
Comment (ob-stripe) [inline src/Stripe.net/Services/PaymentIntents/PaymentIntentTransferDataOptions.cs]: Sorry, nitpicking, but can you change this to something like `"Use PaymentIntent*Options.ApplicationFeeAmount"`, or maybe `"Use ApplicationFeeAmount on the parent options class"`?

`PaymentIntent.ApplicationFeeAmount` refers to the property on the entity class, not the options class, so it's a bit misleading.

Comment (remi-stripe) [reply src/Stripe.net/Services/PaymentIntents/PaymentIntentTransferDataOptions.cs]: yep that's fair. PTAL

**Pull request merged**